frontend/public/components/dashboard/events-card/events-body.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { EventStream, EventStreamList, EventStreamChildProps } from '../../utils/events-stream';
+import { EventItem } from './event-item';
+import { NoEvents, NoMatchingEvents, ErrorLoadingEvents } from '../../events';
+import { LoadingInline } from '../../utils';
+
+const EventsStreamBody: React.FC<Partial<EventStreamChildProps>> = ({ noEvents, noMatches, error, sortedMessages, loading, filteredEvents }) => {
+  if (loading) {
+    return <div className="co-events-card__body-loading"><LoadingInline /></div>;
+  }
+  if (error) {
+    return <ErrorLoadingEvents />;
+  }
+  if (noMatches) {
+    return <NoMatchingEvents allCount={sortedMessages.length} />;
+  }
+  if (noEvents) {
+    return <NoEvents />;
+  }
+
+  return <EventStreamList events={filteredEvents || []} EventComponent={EventItem} scrollableElementId="events-body" />;
+};
+
+export const EventsBody: React.FC<{}> = () => (
```

Comment:
Nit: `EventsCardBody`.

Reply:
other Cards also follow naming pattern without `Card`. I could change it but I'd do it in separate PR for all cards.

Reply:
> other Cards also follow naming pattern without `Card`

OK, let's keep the existing convention then.